pkg/datapath/linux/probes/probes.go

```diff
@@ -422,3 +395,101 @@ func (p *ProbeManager) KernelConfigAvailable() bool {
 
 	return true
 }
+
+// HaveMapType is a wrapper around features.HaveMapType() to check if a certain
+// BPF map type is supported by the kernel.
+// On unexpected probe results this function will terminate with log.Fatal().
+func HaveMapType(mt ebpf.MapType) error {
```

Comment:
I think that `MustHaveMapType()` would be a more canonical way to call this if there are strict consequences for the map type being missing.

Reply:
(next feedback will render this unnecessary)

Reply:
Somehow I just missed the whole `ErrNotSupported` logic, sorry about that. This feedback doesn't really make sense.